### PR TITLE
refactor(excelsheet): Refactor Arrow Array creation

### LINKED ISSUE #31: Refactoring the arrow array's creations
https://github.com/ToucanToco/fastexcel/blob/dfe46bd3cd88b87813cb3eef475fea1b8e8f9b8a/src/types/excelsheet.rs#L48-L80

Maybe we could change this :
- Maybe add a trait
- Maybe a closure
- Maybe something else...

### PR DESCRIPTION
**Pull request by lukapeschke**
closes #31

### CODE REVIEW

**Comment by lukapeschke** [comment]
Not really sure the new form is better :sweat_smile:

File: src/types/excelsheet.rs [hunk 1]
```diff
@@ -141,93 +142,21 @@ impl ExcelSheet {
...
-    })))
-}
-
-fn create_datetime_array(
+fn create_arrow_array<NT, T: ArrowPrimitiveType<Native = NT>, F: Fn(&CalDataType) -> Option<NT>>(
     data: &Range<CalDataType>,
     col: usize,
     offset: usize,
     limit: usize,
+    transformer: &F,
```
Comment (jGundermann) [inline src/types/excelsheet.rs:156]: Is borrowing the function really mandatory ?

File: src/types/excelsheet.rs [hunk 2]
```diff
@@ -247,37 +176,79 @@ impl TryFrom<&ExcelSheet> for RecordBatch {
...
-                },
-            )
+            let array: Arc<dyn Array> = match field.data_type() {
+                ArrowDataType::Boolean => {
+                    Arc::new(BooleanArray::from_iter((offset..limit).map(|row| {
+                        value
+                            .data()
+                            .get((row, col_idx))
+                            .and_then(|cell| cell.get_bool())
+                    })))
```
Comment (jGundermann) [inline src/types/excelsheet.rs:193]: Could we use the helper function here also ?

**Review by jGundermann — COMMENTED** [review]
Nice PR! Some question

File: src/types/excelsheet.rs (hunk 1 above)
Comment (lukapeschke) [reply src/types/excelsheet.rs:156]: Yup. Since it get captured by a closure, it'd be moved if we received it by value

**Pull request closed without merging**